### PR TITLE
More on the two matplotlib interfaces; closes #50

### LINKED ISSUE #50: plotting: two different interfaces
add explanation and possibly a discussion session on why we insist on this

### PR DESCRIPTION
**Pull request by bast**

### CODE REVIEW

File: data-visualization.rst [hunk 1]
```diff
@@ -73,21 +75,23 @@ to show the generated figure in the notebook, we don't need this when running th
...
-   :emphasize-lines: 4
+   :emphasize-lines: 3
 
    # ... rest of the script
 
+   fig.savefig("my-plot.png")
+
    plt.show()
-   plt.savefig("my-plot.png")
+
```
Comment (annefou) [inline data-visualization.rst:92]: It is not only when running python scripts (we haven't dicussed scripts yet). the most common usage at this stage would be to use python console (where plt.show() is then required).

**Comment by bast** [comment]
I am going ahead and merging ...

File: data-visualization.rst [hunk 2]
```diff
@@ -203,6 +203,18 @@ Imagine we wanted to learn how to create a histogram and web searched
 - Change the number of bins.
 - Convert it from pyplot interface to using the object oriented interface.
 
+.. discussion::
+
+   Why did we do this? One day you may want to write functions which wrap
```
Comment (annefou) [inline data-visualization.rst:208]: We also use it to pass additional matplotlib args to fully customize plots.

**Review by annefou — COMMENTED** [review]
I think it is great.

**Pull request merged**